### PR TITLE
Adding parent-partner-ed screens

### PR DESCRIPTION
**Pull request by sree-cfa**
#### 🔗 Pivotal Tracker ticket
PT#186822687

#### ✍️ Description
new screens:
- activities-partner-add-ed-program screen
- activities-partner-ed-program-type screen
- activities-partner-ed-program-name screen
- activities-partner-ed-program-info screen
- activities-partner-ed-program-method screen
- activities-partner-next-class-schedule screen
- activities-partner-class-weekly-schedule screen
- activities-partner-class-hourly-schedule screen
- activities-partner-ed-program-dates screen

Refactoring
- extracted duplicate 
- added/updated some validations

#### 📷 Design reference
[Figma link](https://www.figma.com/file/1oRFY7p6tdGogBJk7Ugle8/IL-CCAP-Full-Flow-(WIP)?type=design&node-id=487-23426&mode=design&t=9iAwAyb6IeEkuqnn-0)

This is the parent education section, the secondary parent is the same except for the copy referring to the partner instead of the primary parent
Screenshot 2024-04-03 at 12 29 24 PM

#### ✅ Completion tasks

- [x] Added relevant tests
- [x] Meets acceptance criteria

### CODE REVIEW

**Comment by coltborg** [comment]
activities-ed-program-info

- [ ] Phone validation is not there
- [ ] Zip validation is not there

File: src/main/resources/templates/fragments/screens/activities-ed-program-method.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  <head th:replace="~{fragments/head :: head(title=#{activities-ed-program-method.header})}"></head>
+  <body>
+  <div class="page-wrapper">
+    <th:block th:replace="~{fragments/toolbar :: toolbar}"/>
+    <section class="slab">
+      <div class="grid">
+        <th:block th:replace="~{fragments/goBack :: goBackLink}"/>
+        <main id="content" role="main" class="form-card spacing-above-35">
+          <th:block th:replace="~{fragments/inputs/headerNameUnderlined :: headerNameUnderlined(text=${programName})}"/>
+          <th:block th:replace="~{fragments/cardHeader :: cardHeader(header=#{activities-ed-program-method.header})}"/>
```
Comment (enyia21) [inline src/main/resources/templates/fragments/screens/activities-ed-program-method.html:13]: Should the program header be the same for all screens?  What happens if we want to make the parent and partner screens have different headers in the future?

File: src/main/resources/templates/gcc/activities-ed-program-method.html [hunk 2]
```diff
@@ -1,54 +1,11 @@
...
-      </main>
-    </div>
-  </section>
-</div>
-<th:block th:replace="~{fragments/footer :: footer}"/>
-</body>
-</html>
+<th:block
+        th:replace="~{fragments/screens/activities-ed-program-method ::
+      screen(
```
Comment (enyia21) [inline src/main/resources/templates/gcc/activities-ed-program-method.html:3]: No header field?

**Review by enyia21 — APPROVED** [review]

File: src/main/resources/templates/fragments/screens/activities-ed-program-method.html (hunk 1 above)
Comment (coltborg) [reply src/main/resources/templates/fragments/screens/activities-ed-program-method.html:13]: If we wanted them different, we would not use this shared screen fragment and move the code back into the individual screens and have them different in each.

File: src/main/resources/templates/gcc/activities-ed-program-method.html (hunk 2 above)
Comment (coltborg) [reply src/main/resources/templates/gcc/activities-ed-program-method.html:3]: Does there need to be a header field?

Comment (enyia21) [reply src/main/resources/templates/gcc/activities-ed-program-method.html:3]: This statement was tied to the [comment](https://github.com/codeforamerica/il-gcc-form-flow/pull/195#discussion_r1561271913) above.

File: src/main/resources/templates/fragments/screens/activities-ed-program-method.html (hunk 1 above)
Comment (coltborg) [reply src/main/resources/templates/fragments/screens/activities-ed-program-method.html:13]: We paired on this, decided to hold off.

**Pull request merged**